cpp/src/join/hash_join.cu

```diff
@@ -60,13 +60,16 @@ namespace {
  *
  * @return The exact size of the output of the join operation
  */
-template <join_kind JoinKind>
```

Comment:
Update docs. 

`JoinKind` is no longer a template parameter

Would this change affect runtime performance?

Reply:
It should not since this parameter is just a quick host-side check and doesn't affect any critical code paths like comparator kernels or hash-map kernels